### PR TITLE
Further reduce APK size

### PR DESCRIPTION
**Pull request by atavism**
Some additional changes for https://github.com/getlantern/lantern-client/pull/1030

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -50,6 +50,11 @@ android {
         exclude 'META-INF/maven/commons-io/commons-io/pom.properties'
         exclude("META-INF/*.kotlin_module")
         exclude 'README.txt'
+
+        jniLibs {
+            useLegacyPackaging true
```
Comment (atavism) [inline android/app/build.gradle:55]: This is the most important change. For whatever reason, using legacy packaging reduces the size of the APK by 100MB!

**Comment by atavism** [comment]
Another change we could make is to use the arm-eabi-strip tool provided by the Android NDK to remove unnecessary debug symbols from native libraries:

```
apkanalyzer -h files list --download-size lantern-installer.apk

55.7MB	/lib/
28.5MB	/lib/x86_64/
1.6MB	/lib/x86_64/libsqlcipher.so
411.7KB	/lib/x86_64/libsentry.so
5.2KB	/lib/x86_64/libsentry-android.so
333.3KB	/lib/x86_64/libopustool.so
109.2KB	/lib/x86_64/libmodpng.so
2.5MB	/lib/x86_64/libmodpdfium.so
289.9KB	/lib/x86_64/libmodft2.so
22.5KB	/lib/x86_64/libjniPdfium.so
14.3MB	/lib/x86_64/libgojni.so
4.8MB	/lib/x86_64/libflutter.so
324.8KB	/lib/x86_64/libc++_shared.so
3.9MB	/lib/x86_64/libapp.so
27.2MB	/lib/arm64-v8a/
1.4MB	/lib/arm64-v8a/libsqlcipher.so
371.7KB	/lib/arm64-v8a/libsentry.so
5.1KB	/lib/arm64-v8a/libsentry-android.so
293.7KB	/lib/arm64-v8a/libopustool.so
99.6KB	/lib/arm64-v8a/libmodpng.so
2.3MB	/lib/arm64-v8a/libmodpdfium.so
269.8KB	/lib/arm64-v8a/libmodft2.so
20.9KB	/lib/arm64-v8a/libjniPdfium.so
13.6MB	/lib/arm64-v8a/libgojni.so
4.7MB	/lib/arm64-v8a/libflutter.so
296.3KB	/lib/arm64-v8a/libc++_shared.so
3.9MB	/lib/arm64-v8a/libapp.so
```

File: Makefile [hunk 2]
```diff
@@ -179,7 +176,7 @@ else ifeq ($(ANDROID_ARCH), all)
 # native libs, 32 bit Intel devices will just emulate ARM.
 # DO NOT ADD x86 TO THIS LIST!!
   ANDROID_ARCH_JAVA := arm64-v8a x86_64
-  ANDROID_ARCH_GOMOBILE := android/arm64,android/amd64
+  ANDROID_ARCH_GOMOBILE := android
```
Comment (atavism) [inline Makefile:179]: "By default, -target=android builds shared libraries for all supported instruction sets" https://pkg.go.dev/golang.org/x/mobile/cmd/gomobile

Comment (jigar-f) [reply Makefile:179]: So basically, Now we will generate go lib for all arch, But include 64 bit while building correct?

File: android/app/build.gradle (hunk 1 above)
Comment (jigar-f) [reply android/app/build.gradle:53]: wooo, That's great.

**Review by jigar-f — APPROVED** [review]
LGTM

**Comment by jigar-f** [comment]
@atavism Do you think, You can change the base branch of this PR and merge this into my #1030 PR, I think by merging these changes, there we might go 3-4 MB more down, I have removed a lot of unused files and resources.

**Comment by jigar-f** [comment]
With the latest changes including from #1031 

The APK size is now 75 MB. Still, there are some class.dex are there, I am trying to find how to get of that compose in android

cc @atavism 

Screenshot 2024-03-28 at 5 23 13 PM

**Comment by atavism** [comment]
@jigar-f Sure, doing that now

**Pull request merged**